### ISSUE TITLE
Crosslingual Embeddings within Flair Framework

### ISSUE THREAD

**Issue by ChessMateK** [flairNLP/flair#1771]
Hello everyone!

I'd like to use a sequence tagger for NER, training, validating and testing it with cross lingual embeddings following [this guide](https://github.com/flairNLP/flair/blob/master/resources/docs/TUTORIAL_7_TRAINING_A_MODEL.md#resuming-training).

In detail I would use [MUSE](https://github.com/facebookresearch/MUSE) and [ELMo Multilingual](https://github.com/TalSchuster/CrossLingualContextualEmb) embeddings.

I found MuseCrosslingualEmbeddings withing flair code and added it through `from flair.embeddings import MuseCrosslingualEmbeddings`. But something strange happens: embeddings are downloaded during training, maybe due the fact that this class was added for a different task as detailed [here](https://github.com/flairNLP/flair/pull/853) and [here](https://github.com/flairNLP/flair/issues/852) by @alanakbik.

I haven't found nothing about ELMo Multilingual.

To sum up, is there any possibility to use MUSE or ELMo Multilingual for NER within the sequence tagger trainer? 
If not, are there other multi/cross lingual embeddings supported by flair than flair and bpe embeddings?

Thank you all.

**Comment by helpmefindaname** [comment]
The flair library works the way, that it downloads the embeddings whenever they are needed/not in the cache already.
So the "something strange" is actual intended behaviour. If you want to download it beforehand, you can do it by beforehands downloading it, by copying the code here: https://github.com/flairNLP/flair/blob/master/flair/embeddings/token.py#L1480

I agree elmo seems to not be implemented, but you can try the flair embeddings:
(see options here: https://github.com/flairNLP/flair/blob/master/flair/embeddings/token.py#L431)
or transformer embeddings: https://huggingface.co/transformers/pretrained_models.html (search for multilingual on that page)

**Comment by alanakbik** [comment]
Hello @ChessMateK yes as @helpmefindaname wrote Flair downloads the models as needed. The MUSE embeddings are a particular case since they consist of many different language-specific embeddings that were mapped onto the same space. So here, the word "cat" and the German translation "Katze" would have theoretically the same embedding vector. Any time MUSE encounters a new language it hasn't seen before, it downloads the appropriate embedding to your machine, which is why it sometimes interrupts the training to wait for it.

The multilingual BPE, Flair and BERT embeddings are different as there is no such guarantee. They can embed any word in any language, but  "cat" and "Katze" might have completely different embedding vectors.
